### DISCUSSION TITLE
Correct size for cursor pointers in the game

### DISCUSSION THREAD

**Discussion by mefistotelis** [swfans/syndwarsfx#222] [General]
I'm now working on making cursor sizes for higher resolutions. Rather than resizing them by the game engine, I'm planning to just prepare sets of cursors for various resolutions.

This made me wonder about the original sizes:

The animated one in menu:
This one is 16x30 pixels, and that seem to be a good choice for the 640x480 menus - it is visible, even though the color isn't very distinct from the menus. But the fact that menus are static after the drawing animation finishes makes it way easier to follow the cursor.

The grey one within the game:
Here things are strange. There's 10x15 cursor, but it's used for both 320x200 and 640x480. Which resolution it matches best? Looks to me like the high-res cursor could be bigger, though doubling the size would probably be too much. 50% bigger, maybe? And the 320x200 one could be a little smaller, not much but 1-2 pixels.

What do you think? what should be the proper size for cursors?

EDIT:
The current cursors can be seen here:
https://github.com/swfans/swars-gfx/tree/main/pointers-64
(the ones for menu are nearby as well)

If anyone feels like scaling any of them, I would accept such contribution.

**Comment by geist22** [comment]
I think the cursor in 640x480 looks fine, and with default settings (unlike higher resolutions that would make the cursor tiny), the port should keep the looks consistent with the original game. Maybe a configurable option for cursor size. I thought scaling the size of the cursor with resolution to keep its size the same compared to the screen would have been a good option.

I don't know how others feel about 320x200, but personally I view it more like a compatibility mode that the developers added so the game could be played on old computers. Some UI elements appear bigger and there are other issues (some of which you've fixed in the port) like the mission objective text in the minimap scrolling much faster. While it's important to have the lowres mode working and playable, I think design issues should first and foremost be based on 640x480 as the "default" mode.

Speaking of 320x200, I've opened a discussion about it recently as I believe you're using the "wrong" aspect ratio for it, did you check it out?

**Reply by mefistotelis** [reply]
> I think the cursor in 640x480 looks fine

Well, I don't know.. for me the cursor looks like smallest acceptable size for the resolution. Having it small helps with precise targeting though. And if I compare it to the cursor I have in my OS, it is slightly larger. But my OS does not have a rotating 3D world where everything moves, so it definitely should be bigger than there.

I'm only planning to have one set of rules which select sprite detail based on resolution. Currently the rule is - if height is >= 400, load the larger ones. The next step after that will be the height of 800. If this size of cursor stays, then it will be used for up to that resolution - ie. 1280×720 will still have the same cursor, which will be way too tiny at that point.

> I thought scaling the size of the cursor with resolution to keep its size the same compared to the screen would have been a good option.

I made a scaling sprites in KeeperFX, but I'm not sure it's a good idea for SW. The game has much "cleaner" feel, and any scaling on such clean sprites will be more visible.

> I've opened a discussion about it recently as I believe you're using the "wrong" aspect ratio for it, did you check it out?

Yes. But I don't see anything actionable on that front.

Will answer there to not pollute this thread. For cursors, Bullfrog used the same ones and ignored the non-square pixels. So, are these made for 320x200, or for 640x480?

**Comment by mefistotelis** [comment]
I prepared some higher res cursors. I'm definitely not an artist, but I did my best.

**Reply by Moburma** [reply]
Ahh, that's really nice. I would agree with @geist22 that I'm not really bothered about the original game mode pointers as they were fine (and I especially agree that 320x200 is basically unimportant/dead at this point), but the new high res pointer in resolutions beyond that of the original game is great, makes it must easier to keep track of the pointer in rainy levels in HD screen modes, good work!

**Comment by mefistotelis** [comment]
It's true there's probably a very narrow range of people who would use 320x200 .. 512x384 modes. I still want it to work, but it's not worth improving.
Still I already prepared the slightly smaller cursors, they look ok to me so can stay. (these are detail=0 pointers)

For 640x480 .. 1024×768 (detail=1), will leave the cursors as-is. They look too small to me, but since you both agree they're ok, I can accept that.

For 1280x800 .. 2048x1536 (detail=2), I prepared cursors by doubling size of previous ones and drawing on outer borders to improve resolution. I feel these probably could still be improved.

For 2560x1600+ (detail=3), we currently don't have cursors. I guess they should be made by just doubling the previous ones. Or we could do a vector-based design, and then use that as a base for specific size cursors.

For detail=2, which one of the following is better?:
![pointer_futuregrey_std](https://github.com/user-attachments/assets/eb080277-bea0-47f0-8493-6244cec27592)
![pointer_futuregrey_std-v2](https://github.com/user-attachments/assets/906337c8-f70a-4124-9262-869a5d0d2249)

**Reply by Moburma** [reply]
Hmm, maybe the second one for me. At the risk of veering off topic, did you ever see what was in MEDIT-0.ANI in the beta demo? I assumed that was some kind of editor mouse cursors, but don't have a way to view it.

**Reply by mefistotelis** [reply]
> did you ever see what was in MEDIT-0.ANI in the beta demo?

From just a quick look I assumed there's nothing interesting.

In general, ANI files are used to create animation frames out of sprites. Then to show a thing on screen you don't set it to a sprite number, but an animation frame number.

Animation frames are sometimes just referencing a sprite and that's it. But they can also reference multiple sprites, cobbling them together, or overlaying with effect. Like agent dying, when his body disappears gradually and is replaced with blood in shape of a person. Or a target frame, combining few sprites at different positions making a rectangle.

Each frame can reference a next frame, making animation.

So at first look, `MEDIT-0.ANI` seem to just contain those - definitions of animations. What differs it from other ANIs is that it contains text names of those animations as well.

**Reply by Moburma** [reply]
> What differs it from other ANIs is that it contains text names of those animations as well.

Ahh, can't believe I missed that. Interestingly it still lists the Female version of the Shady Guy as well...
